app/services/puppet_class_importer.rb

```diff
@@ -94,6 +107,29 @@ def updated_classes_for(environment)
     ]
   end
 
+  # Gives back the classes ignored for a given environment
+  #
+  # Params:
+  #  * ++environment++:: {String} name of the environment
+  #
+  def ignored_classes_for(environment)
+    classes = proxy.classes(environment)
+    classes.keys.select do |class_name|
+      ignored_class?(class_name)
+    end
+  end
+
+  # Returns true when the class name matches any pattern in ignored_classes
+  #
+  # Params:
+  #  * ++class_name++:: {String} containing the class to be checked
+  #
+  def ignored_class?(class_name)
+    !ignored_classes.select do |filter|
+      filter =~ class_name
```

Comment:
This duplicates the check in `actual_classes` but without the safety check for Regexp - it should be shared code.